week-2/Homework/extra/4-radio-stations.js

```diff
@@ -25,57 +34,78 @@
  * - Sort the stations by low - high e.g. 1,2,3,4,5
  * - Return only the frequencies that are radio stations.
  */
-// `getStations` goes here
+
+function getStations(arrFreq) {
+  let stationsList = []
+  for (let i = 0; i < arrFreq.length; ++i)
+    if (isRadioStation(arrFreq[i])) {
+      stationsList.push(arrFreq[i])
+    }
+  stationsList.sort((x, y) => x - y)
+  console.log(stationsList)
+  return stationsList
+}
 
 /**
  * Lastly, let's make a function for people to use.
- * When this runs, it should use the above two functions to log to the `console`
- * Call this function `searchRadioWaves`.
+@@ -34,48 +53,58 @@
  * - "🎶 station found at ${frequency}, enjoy!"
  * - "No station found at ${frequency}, moving on"
  */
 // `searchRadioWaves` goes here
-
+function searchRadioWaves() {
+  let frequenciesArray = getAllFrequencies()
+  let stationsArray = getStations(frequenciesArray)
+  for (let y = 0; y < frequenciesArray.length; ++y) {
+    if (stationsArray.includes(frequenciesArray[y])) {
+      console.log(`🎶 station found at ${frequenciesArray[y]}, enjoy!`)
+    } else {
+      console.log(`No station found at ${frequenciesArray[y]}, moving on`)
+    }
+  }
+}
+searchRadioWaves()
 /* ======= TESTS - DO NOT MODIFY ======= */
-
 function isRadioStation(frequency) {
-  // This is a way of storing the random frequency array inside a 
+  // This is a way of storing the random frequency array inside a
+  // This is a way of storing the random frequency array inside a
   // function as storing it outside is bad practice. Don't worry if you
   // don't understand some bits! You're more than welcome to ask questions
   // at any time. :)
+
   if (!this.stations) {
-    const stationCount = 4;
+    const stationCount = 4
     const availableStations = new Array(stationCount)
       .fill(undefined)
       .map((_, i) => {
-        return Math.floor(Math.random() * (10800 - 8700 + 1) + 8700) / 100;
-      });
+        return Math.floor(Math.random() * (10800 - 8700 + 1) + 8700) / 100
+      })
 
-    this.stations = availableStations;
+    this.stations = availableStations
   }
 
-  return this.stations.includes(frequency);
+  return this.stations.includes(frequency)
 }
 
-const assert = require("assert");
-
+const assert = require('assert')
```

Comment:
I'm really glad to see you experimenting with the `require` statement here! Did you put this in yourself @byohannes ?

Reply:
@sansaid it was given readily within the test code but I have researched about what it does as // Generate an AssertionError to compare the error message